net/Socket.hpp

```diff
@@ -395,9 +464,17 @@ class Socket
 #endif
     }
 
-private:
     std::string _clientAddress;
+    const Type _type;
+    unsigned int _clientPort;
     const int _fd;
+    /// True if this socket is open.
+    bool _open;
```

Comment:
We need to care about size here. Can we use bit-fields to pack _type, _clientPort, and _open into the same space ?

Reply:
I often use bitfields when having considerable flags indeed.
However, each socket is acompanied by much larger buffers (1-8k) and I doubt 3 x sizeof(int) .. but well, keeping it in mind that its OK for you. An artifical encoding of different types (int + bool) is however .. overkill?
Sure, if we would talk about an object handle, function-ptr or any short lived type in quantities of >= 100k w/o any other side-effects or memory costs otherwise (sure) .. but I don't see it here.
Was thinking about the added memory costs of Socket with added statistics, correct. But as described above .. I doubt this is material here. I could be wrong ofc.

Reply:
Why not lift `_closed` from `StreamSocket` to its parent, here?

At a minimum we should review `StreamSocket::_closed`, now that we have `_open`. Otherwise, horrors when these two go out of sync and the left hand fights the right.

Of course it's also good when adding such common flags to review how we dealt with them for so long, to avoid redundancy and unify the design. Perhaps `_open` was enough and this is redundant. Perhaps not. Worth sharing your analysis either way.

Reply:
StreamSocket::_closed has been moved to Socket::_open